forest/src/cli/genesis_cmd.rs

```diff
@@ -45,6 +63,62 @@ impl GenesisCommands {
                     }
                 }
             }
+            Self::AddMiner {
+                genesis_path,
+                preseal_path,
+            } => {
+                if let Err(err) = add_miner(genesis_path.to_string(), preseal_path.to_string()) {
+                    println!("Cannot add miner(s), error: {}", err)
+                };
+            }
+        }
+    }
+}
+
+fn add_miner(genesis_path: String, preseal_path: String) -> Result<(), Box<dyn std::error::Error>> {
+    let mut genesis_str = String::new();
+    File::open(&genesis_path)?.read_to_string(&mut genesis_str)?;
+    let mut template: GenesisTemplate = serde_json::from_str(&genesis_str)?;
+
+    let mut preseal_str = String::new();
+    File::open(preseal_path)?.read_to_string(&mut preseal_str)?;
+    let miners: HashMap<String, Miner> = serde_json::from_str(&preseal_str)?;
+
+    for (miner_address_str, miner) in miners.into_iter() {
+        println!("Adding miner {} to genesis template", miner_address_str);
```

Comment:
Sorry if I didn't mention on the other changes or CLI subcommands, but I think it's better to use the logger (it's already setup so all you have to do is replace `println!` with `info!` using `log::info`. (Can use any logging level you see necessary though

Reply:
switched all `println!` to `info` and `warn`, thank you!